File: Challenge_1.py
# -*- coding: utf-8 -*-
"""

@author: Ben
"""

import scipy.integrate as sci
import numpy as np
from matplotlib import pyplot as pl
from math import sqrt

a = 2   # amplitude 
N = 1000   # number of points
m = 1
x = np.linspace(0,a,N,endpoint=False)
f = lambda x:sqrt((8*m)/(a**4-x**4)) 
f_array = np.vectorize(f)
f_array = f_array(x)

t1 = sci.trapz(f_array,x)   #trapezoidal method

t2 = sci.simps(f_array,x)   #simpsons method

print ("t_trap , t_simp", t1,t2)

t = []
a = np.linspace(.01,4,1000)

for i in a:
    f = lambda x:sqrt((8*m)/(i**4-x**4))
    x = np.linspace(0,i,N,endpoint=False)
    f_array = np.vectorize(f)
    f_array = f_array(x)
    t.append(sci.trapz(f_array,x)) 
 

pl.plot(a,t)
pl.xlabel('Amplitude')
pl.ylabel('Period')

File: Challenge_1.m
clear;

m = 1;
amp = 2;
x = linspace(0,amp - 0.0001,10000);
n = length(x);
f = @(x) sqrt((8*m)./(amp.^4 - x.^4));
y = f(x);
t1 = integral(f,x(1),x(n));
t2 = trapz(x,y);
fprintf('MATLAB Integral Function: t = %f \nTrapezoid Method: t = %f',t1,t2);

f = @(x,a) sqrt((8*m)./(a.^4 - x.^4));
a = linspace(0.01,4,10000);
n = length(a);

t = [];

for i = a(1:n)
x = linspace(0,i - 0.0001,10000);
t = [t,trapz(x,f(x,i))];
end

plot(a,t,'-');
xlabel('Amplitude');
ylabel('Period');
